node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -382,26 +385,40 @@ func (node *testOnlyProcessingNode) collectClosableComponents(apiInterface APICo
 	}
 }
 
-// SetState will set the provided state for the given address
-func (node *testOnlyProcessingNode) SetState(address []byte, keyValueMap map[string]string) error {
+// SetKeyValueForAddress will set the provided state for the given address
+func (node *testOnlyProcessingNode) SetKeyValueForAddress(address []byte, keyValueMap map[string]string) error {
+	userAccount, err := node.getUserAccount(address)
```

Comment:
what happens if the address is not in the shard maintained by this testOnlyProcessingNode instance?